### PR TITLE
Add OIDC integration for Lieutenant communication

### PR DESCRIPTION
**Pull request by glrf**
This PR introduces a `commodore login` command that fetches an `id_token` from the configured IDP and stores it in a cache at `~/.cache/commodore/token`. This cache is used to lookup API tokens if no token is explicitly set.

You can test the login with `lieutenant-dev` using 

```
export COMMODORE_API_URL="https://api-dev.syn.vshn.net"
export COMMODORE_OIDC_CLIENT=syn-lieutenant-dev 
export COMMODORE_OIDC_DISCOVERY_URL=https://id.test.vshn.net/auth/realms/VSHN-main-dev-realm/.well-known/openid-configuration

commodore login
commodore catalog compile c-glrf-test
```

I'm still  not very happy with the tests..

Things I still want to do, but not in this PR:
* Automatically call `commodore login` if there is no token in the cache / token is expired
* Return the client-id and discovery URL from lieutenant. Which would mean we would be able to just call
`COMMODORE_API_URL="https://api-dev.syn.vshn.net" commodore login`

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update tests.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

**Review by simu — CHANGES_REQUESTED** [review]
LGTM overall. See inline comments for potential improvements.

There's more documentation (e.g. `explanation/running-commodore.adoc`) which will need to be updated at some point, but that's not urgent yet.

File: commodore/tokencache.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import os
+from typing import Optional
+from pathlib import Path
+import json
+
+cache_name = f"{Path.home()}/.cache/commodore/token"
```
[the remark this answers is not in the document]
Comment (glrf) [reply commodore/tokencache.py:6]: Thanks! Yes that's a lot better, but I didn't know where to find it

File: commodore/login.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    if config.oidc_discovery_url is None:
+        raise click.ClickException("Required OIDC discovery URL not set")
+
+    client = WebApplicationClient(config.oidc_client)
+    idp_cfg = get_idp_cfg(config.oidc_discovery_url)
+
+    server = OIDCCallbackServer(client, idp_cfg["token_endpoint"], config.api_url)
+    server.start()
+
+    # That's racy, but it should work most of the time and if not the browser should retry
```
[the remark this answers is not in the document]
Comment (glrf) [reply commodore/login.py:209]: Good idea. I tried to find the status of the server through the server object and failed. I'll try to implement the health endpoint.

**Review by simu — APPROVED** [review]
LGTM.

I had an idea while showering this morning: We could save the Keycloak client id and discovery URL in the token file until we can fetch them from Lieutenant. That way, you only have to specify them once for any given Lieutenant instance, and Commodore can extract them from the token file, if they're not given for subsequent calls based on the provided API URL.

**Comment by glrf** [comment]
> I had an idea while showering this morning: We could save the Keycloak client id and discovery URL in the token file until we can fetch them from Lieutenant. That way, you only have to specify them once for any given Lieutenant instance, and Commodore can extract them from the token file, if they're not given for subsequent calls based on the provided API URL.

I suppose that make using it a bit more convenient, but IMO not by a lot, at least for the way I use commodore right now (sourcing an env file with URL and TOKEN)

Might be a good intermediate solution. Let's see what we manage to do this sprint. I'll merge this for now.

**Pull request merged**